build_web_compilers/pubspec.yaml

```diff
@@ -1,10 +1,10 @@
 name: build_web_compilers
-version: 2.11.0
+version: 2.12.0
 description: Builder implementations wrapping Dart compilers.
 homepage: https://github.com/dart-lang/build/tree/master/build_web_compilers
 
 environment:
-  sdk: ">=2.8.1 <3.0.0"
+  sdk: ">=2.9.0-10.0.dev <3.0.0"
```

Comment:
cc @natebosch @leafpetersen 

Note that this will potentially cause problems with the NNBD tech preview...this package will now be opted in if the experiment is enabled. Concretely that would affect builder migrations, no builder can run `pub run --enable-experiment=non-nullable build_runner <command>` with that stable sdk (or the current dev), they would have to update to the later `2.10.0-x.x.dev` release.

This highlights exactly the scenario we were concerned about last week. Not having an explicit language version field separate from the sdk version is problematic for cases like these. Imagine NNBD was already released, we would be blocked on our NNBD migration in order to depend on this functionality, but this package will never migrate to NNBD most likely, or certainly not for a very long time (the priority is extremely low to do that for this package, and it has a lot of transitive deps it would anyways be blocked on). At the same time it regularly depends on new compiler functionality. The workaround is language version comments in _every single file_ but gross.

Reply:
FWIW I don't mind language version comments in every file.

Reply:
That is a pretty large technical debt burden and not something we are likely to remember to continue doing if we add new files though? We won't be broken or notice since we aren't enabling the experiment...

Reply:
Noted.  The --enable-experiment has become more of a public facing thing than I at least had originally anticipated.  Short term, I don't think much we can do about it. Not sure what the right resolution is long term.  I guess one possibility is to require something extra in the pubspec to opt in (at least during the experiment phase) or equivalently use something besides the min sdk version to opt in.  The latter ship feels like it has sailed though.